src/ch02/hash-solidity-compatible.md

```diff
@@ -0,0 +1,11 @@
+# Hash Solidity Compatible
+
+This contract demonstrates Keccak hashing in Cairo to match Solidity's keccak256. While both use Keccak, their endianness differs: Cairo is little-endian, Solidity big-endian. The contract achieves compatibility by hashing in big-endian using `keccak_u256s_be_inputs`, and reversing the bytes of the result with `u128_byte_reverse`.
+
+For example:
+
+```rust
+{{#include ../../listings/ch02-advanced-concepts/hash_solidity_compatible/src/contract.cairo}}
+```
+
+Visit contract on [Voyager](#) or play with it in [Remix](https://remix.ethereum.org/?#activate=Starknet&url=https://github.com/NethermindEth/StarknetByExample/blob/main/listings/ch02-advanced-concepts/hash_solidity_compatible/src/contract.cairo).
```

Comment:
If there's no deployed instance of the contract you can remove the "Visit contract on Voyager"